pkg/commands/chgm/chgm.go

```diff
@@ -0,0 +1,147 @@
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+)
+
+// these type aliases are here to make the types unique and unambiguus when using inside the struct
+type AwsClient = aws.Client
+type OcmClient = ocm.Client
+
+// ChgmProvider should have all of the functions that ChgmService is implementing
+type ChgmProvider struct {
+	// having awsClient and ocmClient this way
+	// allows for all of the method receivers defined on them to be passed into the parent struct,
+	// thus making it more composable than just having each func redefined here
+	//
+	// a different solution is to have the structs have unique names to begin with, which makes the code
+	// aws.AwsClient feel a bit redundant
+	// TODO: decide if the client types should be unique or not
+	AwsClient
+	OcmClient
+}
+
+// This will generate mocks for the interfaces in this file
+//go:generate mockgen --build_flags=--mod=readonly -source $GOFILE -destination ./mock/interfaces.go -package mock
+
+// ChgmService will wrap all the required commands the client needs to run it's operations
+type ChgmService interface {
+	// AWS
+	ListStopInstancesEvents() ([]*cloudtrail.Event, error)
+	ListRunningInstances(infraId string) ([]*ec2.Instance, error)
+	ListStoppedInstances(infraId string) ([]*ec2.Instance, error)
+	// OCM
+	GetClusterDeployment(clusterID string) (*hivev1.ClusterDeployment, error)
+	GetClusterInfo(identifier string) (*v1.Cluster, error)
+}
+
+// Client is an implementation of the Interface, and adds functionality above it
+// to create 'Client' you can use a mock, or fill it with
+type Client struct {
+	ChgmService
+}
+
+// isUserAllowedToStop will verify if a user is allowed to stop instances.
+// as this is the thing that might change the most it is at the top.
+// additionally it is private as I don't see anyone using this outside of AreAllInstancesRunning
+func isUserAllowedToStop(username, infraID string) bool {
+	//TODO: what is the best ordering for this? (from the most common to the most rare)
+
+	// operatorIamNames will hold all of the iam names that are allowed to stop instances
+	// TODO: (remove when there is more than one item) holds only one item to allow adding IAM stuff later
+	// pulled by:
+	// 1. logging (via osdctl account cli) into an aws cluster
+	// 2. running the command "aws iam list-users --query 'Users[?starts_with(UserName,`<INFRA_ID>`)].UserName'"
+	// 3. trimming the infra id from the front and the uuid from the back
+	// 4. curate the list down until you have only the required api's
+	operatorIamNames := []string{
+		"openshift-machine-api-aws",
+	}
+	for _, operatorIamName := range operatorIamNames {
+		// I added an additional '-' in the end to make sure it's the same as close as I can to the aws username (cannot guess the uuid)
+		operatorIamName = fmt.Sprintf("%s-%s-", infraID, operatorIamName)
+		if strings.HasPrefix(username, operatorIamName) {
+			return true
+		}
+	}
+
+	switch {
```

Comment:
I would do an if out of this:

```go
if strings.HasPrefix(username, "osdManagedAdmin-") {
    return true
}
return false
```

It's easier to read and "flatter".

Reply:
Yeah, switch case was because I had more cases.
Will change to your way

Reply:
Ah okay. In that case a switch case would be the case to choose :D :D :D